src/readblock.c

```diff
@@ -0,0 +1,299 @@
+static int at_decorative_element (ffio *F) {
+/****************************************************************************************
+    A decorative element consists of a line of at least 5 consecutive identical chars,
+    starting at buffer position 0:
+    "-----", "=====", "*****", etc.
+
+    A decorative element serves as a end delimiter for the current element, and
+    continues until a gie command verb is found at the start of a line
+****************************************************************************************/
+    int i;
+    char *c;
+    if (0==F)
+        return 0;
+    c = F->next_args;
+    if (0==c)
+        return 0;
+    if (0==c[0])
+        return 0;
+    for (i = 1; i < 5; i++)
+        if (c[i]!=c[0])
+            return 0;
+    return 1;
+}
+
+
+/***************************************************************************************/
+static const char *at_tag (ffio *F) {
+/****************************************************************************************
+    A start of a new command serves as an end delimiter for the current command
+****************************************************************************************/
+    size_t j;
+    for (j = 0;  j < F->n_tags;  j++)
+        if (strncmp (F->next_args, F->tags[j], strlen(F->tags[j]))==0)
+            return F->tags[j];
+    return 0;
+}
+
+
+/***************************************************************************************/
+static int at_end_delimiter (ffio *F) {
+/****************************************************************************************
+    An instruction consists of everything from its introductory tag to its end
+    delimiter.  An end delimiter can be either the introductory tag of the next
+    instruction, or a "decorative element", i.e. one of the "ascii art" style
+    block delimiters typically used to mark up block comments in a free format
+    file.
+****************************************************************************************/
+    if (F==0)
+        return 0;
+    if (at_decorative_element (F))
+        return 1;
+    if (at_tag (F))
+        return 1;
+    return 0;
+}
+
+
+/***************************************************************************************/
+static int nextline (ffio *F) {
+/****************************************************************************************
+    Read next line of input file
+****************************************************************************************/
+    F->next_args[0] = 0;
+    if (0==fgets (F->next_args, (int) F->next_args_size - 1, F->f))
+        return 0;
+    if (feof (F->f))
+        return 0;
+    pj_chomp (F->next_args);
+    F->next_lineno++;
+    return 1;
+}
+
+/***************************************************************************************/
+static int locate_tag (ffio *F, const char *tag) {
+/****************************************************************************************
+    find start-of-line tag
```

Comment:
So I think this is consuming lines until the tag is reached and returns true.  And if it runs out of file without finding it, it returns a false.  Yes?  Please add what it does to the comment

Reply:
Done